### PR TITLE
update dr notice query 

### PR DESCRIPTION
**Pull request by adamrunner**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [ ] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/186375269

# A brief description of the changes

Current behavior: We currently use a query that only looks at the `eligibility_determination.outstanding_verification_earliest_due_date` field; this does not appear to capture all evidences and all due dates for individual family members. 

New behavior: The new query will look at all of the due dates on all of the evidences (specifically for `aptc_csr_credit` and `aca_individual_market_eligibility` eligibilities) and will also only look at `evidence_states.status` fields that are `outstanding` 

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name: `TRIGGER_DOCUMENT_REMINDER_NOTICES_AT_INDIVIDUAL_LEVEL_IS_ENABLED`

- [ ] DC
- [x] ME 

Feature flag change in k8s repo: https://github.com/ideacrew/cme_k8s/pull/109

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

File: app/models/family.rb [hunk 1]
```diff
@@ -143,8 +146,8 @@ class Family
```
Comment (ipublic) [inline app/models/family.rb:146]: Add an index for eligibility_determination.subjects.eligibility_states.evidence_states.due_on

File: app/models/family.rb [hunk 2]
```diff
@@ -278,7 +281,14 @@ class Family
   }
 
   scope :outstanding_verifications_expiring_on, lambda {|date|
-    where(:"eligibility_determination.outstanding_verification_earliest_due_date" => date.beginning_of_day)
+    where({
```
Comment (ipublic) [inline app/models/family.rb:284]: Does this criteria include only subjects with evidence state for enrollment == true ?

Recommend making that a separate scope --outstanding_verifications_subjects_enrolled -- that can be chained here and into other scopes (and make it more obvious that non-enrolled family members are ignored)

**Review by ipublic — COMMENTED** [review]
The Family eligibility stores a single verification_earliest_due_date value.  As i recall, the eligibility code sets this value by comparing each candidate subject family member's due_date and selecting the minimum value.  Note that to be a candidate, the subject family member must be enrolled (each subject family member has an enrollment state evidence).

Presume the breakdown occurs when there is one or more subject family members with a due_date != verification_earliest_due_date Those other dates don't match query criteria and trigger a notice.  The solution thus changes query criteria to compare each candidate subject family member's due_date for match with passed date value parameter.

File: app/models/family.rb (hunk 2 above)
Comment (adamrunner) [reply app/models/family.rb:284]: >Does this criteria include only subjects with evidence state for enrollment == true ?

Currently, no. Also not sure which attributes we should be checking on the `EvidenceState` or `EligibilityState` models - there is an `is_eligble` attribute, is that what you were referring to? 

There does not appear to be an `enrollment` attribute on either of those models.

**Review by TristanB17 — APPROVED** [review]
LGTM!

**Review by scaustin34 — APPROVED** [review]

**Pull request merged**